pkg/apis/apps/v1beta2/defaults.go

```diff
@@ -105,16 +88,9 @@ func SetDefaults_StatefulSet(obj *appsv1beta2.StatefulSet) {
 // - RevisionHistoryLimit set to 10 (not set in extensions)
 // - ProgressDeadlineSeconds set to 600s (not set in extensions)
 func SetDefaults_Deployment(obj *appsv1beta2.Deployment) {
-	// Default labels and selector to labels from pod template spec.
 	labels := obj.Spec.Template.Labels
-
-	if labels != nil {
-		if obj.Spec.Selector == nil {
-			obj.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
-		}
-		if len(obj.Labels) == 0 {
-			obj.Labels = labels
-		}
+	if labels != nil && len(obj.Labels) == 0 {
```

Comment:
Delete this. You are setting the labels of the Deployment to the labels of the PodTemplateSpec. You do not want to do this.